src/transform.rs

```diff
@@ -161,6 +183,70 @@ impl<T: en::Num> Transform<T> {
             inv_det * (self.m31 * self.m12 - self.m11 * self.m32),
         ))
     }
+
+    pub fn decompose(&self) -> (Vector<T>, Angle<T>, Angle<T>, Vector<T>)
+    where
+        T: en::Float,
+    {
+        let mut y_flip = T::one();
+        let translation = Vector::new(self.m31, self.m32);
+
+        let rotation_x = Angle::from_xy(self.m11, self.m12);
+        let rotation_y = Angle::from_xy(self.m21, self.m22);
+        let mut skew = (rotation_y - rotation_x).normalize();
+
+        // The reflection axis is undefined, assume Y
+        // i.e. rotating   0º and flipping X
+        //    = rotating 180º and flipping Y
+        if skew.radians() < T::zero() {
+            skew = Angle::from_radians(skew.radians().abs());
+            y_flip = -T::one();
+        }
+
+        // 90º XY angle = 0 skew
+        skew -= Angle::FRAC_PI_2();
+
+        let scale = Vector::new(
+            Vector::new(self.m11, self.m12).magnitude(),
+            Vector::new(self.m21, self.m22).magnitude() * y_flip,
+        );
+
+        (translation, rotation_x, skew, scale)
+    }
+
+    pub fn from_decomposition(
+        translation: Vector<T>,
+        rotation: Angle<T>,
+        skew: Angle<T>,
+        scale: Vector<T>,
+    ) -> Self
+    where
+        T: en::Float,
+    {
+        let c1 = rotation.cos();
```

Comment:
I feel this would be cuter:
```suggestion
        let (s1, c1) = rotation.sin_cos();
```
(especially for the `s2`/`c2` below)